src/prefix-tree/trie.ts

```diff
@@ -75,7 +75,7 @@ export function insert(nodes: Nodes, node: Node, word: string, docId: string): v
 
     if (i === wordLength - 1) {
       node.end = true;
-      node.docs.push(docId);
+      node.docs.push({ id: docId, tf: frequency });
```

Comment:
This looks to me like an associative structure in disguise: each entry is basically a key (the ID) and a value (the frequency). Why not using an object or a map instead? An array also forces you to use `findIndex` below, which is linear, instead of a constant time lookup in an object.